numpy/ma/tests/test_extras.py

```diff
@@ -64,6 +64,18 @@ def test_masked_all(self):
         control = array([[(1, (1, 1))]], mask=[[(1, (1, 1))]], dtype=dt)
         assert_equal(test, control)
 
+    def test_masked_all_with_dtype_str(self):
+        # Test masked_all works with nested array with dtype of an 'object'
+        # refers to issue #15895
+        my_dtype = np.dtype([('b', [('c', object)], (1,))])
```

Comment:
The other issue case I identified above is when the dtype is `my_dtype = np.dtype([('b', (object, (1,)))])`

Reply:
Just to make sure, if we try to access arr['b'], the expected _fill_value for it is still () right?